### PR TITLE
test: remove unnecessary pod ID lookup

### PR DESCRIPTION
**Pull request by whynowy**
Explain what this PR does.

### CODE REVIEW

File: test/fixtures/when.go [hunk 1]
```diff
@@ -42,33 +42,12 @@ type When struct {
...
-		}
-		if len(podList.Items) == 0 {
-			w.t.Fatalf("No running pod found in pipeline %s, vertex: %s", pipelineName, vertexName)
-		}
-		w.vertexToPodIpMapping[labelSelector] = podList.Items[0].Status.PodIP
-	}
-
-	// There could be a rare corner case when a previous added pod gets replaced by a new one, making the mapping entry no longer valid.
-	// Considering current e2e tests are all lightweight, we assume no such case for now.
-	SendMessageTo(w.vertexToPodIpMapping[labelSelector], vertexName, req)
```
Comment (KeranYang) [inline test/fixtures/when.go:71]: Thanks for making the change Derek. I was not aware of the headless service. nit: if you search for the keyword `podIp` across the code base, you will see that I used it a lot. I think with this change, we can update it to maybe `headlessEndpoint` or another better name.

**Review by KeranYang — APPROVED** [review]

**Review by vigith — APPROVED** [review]
